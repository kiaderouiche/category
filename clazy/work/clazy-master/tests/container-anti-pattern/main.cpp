#include <QtCore/QSet>
#include <QtCore/QHash>
#include <QtCore/QMap>
#include <QtCore/QList>
#include <QtCore/QVector>

void test()
{
    QSet<int> set;
    QMap<int,int> map;
    QHash<int,int> hash;
    QList<int> list;
    QVector<int> vec;

    vec.toList().count(); // Warning
    map.values()[0]; // Warning
    int a = hash.keys().at(0); // Warning
    a = map.keys().at(0); // Warning
    list.toVector().size(); // Warning
}

void testRangeLoop()
{
    QHash<int,int> hash;
    for (auto i : hash.values()) {} // Warning
    for (auto i : hash.keys()) {} // Warning
    for (auto i : hash) {} // OK
    foreach (auto i, hash.keys()) {} // Warning
    foreach (auto i, hash.values()) {} // Warning
    foreach (auto i, hash) {} // OK
}
