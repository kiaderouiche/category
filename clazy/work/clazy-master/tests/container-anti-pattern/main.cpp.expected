container-anti-pattern/main.cpp:15:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:16:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:17:13: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:18:9: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:19:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:25:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:26:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:28:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
container-anti-pattern/main.cpp:29:5: warning: allocating an unneeded temporary container [-Wclazy-container-anti-pattern]
